src/diagnostics/wizardSteps/GetUnoccupiedInstance.ts

```diff
@@ -73,4 +76,25 @@ export class GetUnoccupiedInstance extends WizardStep {
             throw new Error(errorMessage);
         }
     }
+
+    private getCurrentUserName(): string {
+        const azureAccountExtension: vscode.Extension<AzureAccount> | undefined = vscode.extensions.getExtension<AzureAccount>('ms-vscode.azure-account');
+        const sessions = azureAccountExtension.exports.sessions;
+
+        const currentTenentId = this._wizard.uiTreeItem.tenantId;
+
+        if (!sessions || sessions.length <= 0) {
+            throw new Error('Cannot find active sessions');
+        }
+
+        const filteredSessions = sessions.filter((session) => {
+            return session.tenantId === currentTenentId;
+        });
+
+        if (!filteredSessions || filteredSessions.length !== 1) {
+            throw new Error(`Cannot locate unique session with tenent ID ${currentTenentId}`);
```

Comment:
this error message may be more useful if you included the count of sessions found.  Also, I think that filter will return an empty array if nothing matches. (you should dbl check), but if so you can simplify things a bit.

Reply:
Thanks for the suggestion. I suppose this whole chunk of work will be removed during the rework.